apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/EnumMap.cs

```diff
@@ -49,32 +60,28 @@ internal static class EnumMap
     /// </summary>
     internal static class EnumMap<T> where T : struct
     {
-        private static readonly Dictionary<string, T> s_stringToValue = new Dictionary<string, T>();
-        private static readonly Dictionary<T, string> s_valueToString = new Dictionary<T, string>();
-        private static readonly Dictionary<int, string> s_intValueToString =
+        private static readonly Dictionary<string, T> s_stringToValue =
             typeof(T).GetTypeInfo().IsDefined(typeof(FlagsAttribute))
-            ? new Dictionary<int, string>() : null;
+            ? null : new Dictionary<string, T>();
+        private static readonly Dictionary<T, string> s_valueToString = 
+            s_stringToValue == null ? null : new Dictionary<T, string>();
 
         static EnumMap()
         {
-            foreach (var field in typeof(T).GetTypeInfo().DeclaredFields.Where(f => f.IsStatic))
+            if(s_stringToValue == null)
```

Comment:
Rather than put the conditional in the initializer and then switch here, I'd check the flags attribute in this "if" condition, throw if necessary, and then assign to s_valueToString after the if statement. I think that'll make it simpler to read - basically everything necessary is in the initializer. (Ditto for the FlagsEnumMap)

Reply:
Doing